components/BottomNavigation/src/MDCBottomNavigationBarController.m

```diff
@@ -30,11 +33,24 @@ - (instancetype)init {
     _content = [[UIView alloc] init];
     _viewControllers = @[];
     _selectedIndex = NSNotFound;
+
+#ifdef DEBUG
```

Comment:
I believe this would be MDC-iOS's first introduction of DEBUG-only code. I'm wary of such an introduction since it makes it harder to determine behavior between debug and production builds.

As an alternative, what do you think of using try-catch and NSAsserting on failure + logging a message?  That would make the app crash in debug failures, not crash in production, and have the same code either way.